vb-fe/src/components/roomModal/RoomModal.tsx

```diff
@@ -63,63 +80,49 @@ export const CreateRoom: FunctionComponent<Props> = ({ open, handleError }) => {
     return true;
   };
 
-  /**
-   * Submits the rom that emits an event with the room details
-   * and closes the modal
-   * @param e form data - has all the details of a room
-   */
-  const onSubmit = (e: any) => {
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     if (validateForm()) {
-      dispatch(createRoomAction(room));
-      setRoom(initialRoomValues);
+      if (modalType === SystemConstants.CREATE) {
+        dispatch(createRoomAction(room));
+      } else if (modalType === SystemConstants.EDIT && currentRoom) {
+        dispatch(updateRoomAction(room, currentRoom._id));
+      }
     }
   };
 
-  /**
-   * Updates the room name and description on change
-   *
-   * @param e event to update name and description
-   */
-  const handleChange = (e: any) => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const { name, value } = e.target;
 
     if (name === "name" && value) setRoom({ ...room, error: false });
-
     setRoom({ ...room, [name]: value });
   };
 
-  /**
-   * Adds a tag emitted from the Select component to the
-   * room state
-   *
-   * @param tag tag that has been added
-   */
   const handleUpdateTags = (tags: Tag[]) => {
     setRoom({ ...room, tags });
   };
 
-  const onClose = () => {
-    dispatch({ type: SystemConstants.CREATE_ROOM_MODAL, payload: false });
+  const handleClose = () => {
+    dispatch({ type: SystemConstants.SET_ROOM_MODAL, payload: { isOpen: false, type: null } });
   };
 
   return (
-    <Modal isOpen={open} onClose={onClose} size="2xl" isCentered autoFocus={false} closeOnOverlayClick={false}>
+    <Modal isOpen={open} onClose={handleClose} size="2xl" isCentered autoFocus={false} closeOnOverlayClick={false}>
       <ModalOverlay bgColor="blackAlpha.800" />
-      <ModalContent bg="gray.800" p={{ base: "0", md: "3" }} py="2">
+      <ModalContent bg="gray.800" p={{ base: "0", md: "3" }} p2="2">
         <ModalHeader display="flex" alignItems="center" w="100%" justifyContent="space-between">
-          <Heading size="lg">Create a room</Heading>
+          <Heading size="lg">{headerText}</Heading>
           <ModalCloseButton position="relative" top="none" right="none" />
         </ModalHeader>
         <ModalBody mt={4}>
-          <form id="create-form" onSubmit={onSubmit}>
+          <form id="create-form" onSubmit={handleSubmit}>
             <FormControl>
               <FormLabel htmlFor="name">Room name</FormLabel>
               <InputGroup display="flex" flexDir="column">
                 <Input
                   isInvalid={room.error}
                   placeholder="Enter a catchy room name!"
-                  onChange={handleChange}
+                  onChange={(e) => handleChange(e)}
```

Comment:
shouldn't this be `onChange={handleChange}`?

Reply:
Yeah, good call - I always do (e) => initially so that I can check the `e` type and plug it into the function args